src/networking.c

```diff
@@ -2546,7 +2563,7 @@ void readQueryFromClient(connection *conn) {
     if (c->reqtype == PROTO_REQ_MULTIBULK && c->multibulklen && c->bulklen != -1
         && c->bulklen >= PROTO_MBULK_BIG_ARG)
     {
-        ssize_t remaining = (size_t)(c->bulklen+2)-sdslen(c->querybuf);
+        ssize_t remaining = (size_t)(c->bulklen+2)-(sdslen(c->querybuf)-c->qb_pos);
```

Comment:
was that a plain bug? or a change needed because processInputBuffer doesn't always do sdsrange?

Reply:
it's a bug but never take effect before this PR, because processInputBuffer always did sdsrange, but now the `qb_pos` may not be 0 if it's master client.